core/blockstm/executor.go

```diff
@@ -0,0 +1,241 @@
+package blockstm
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type ExecResult struct {
+	err      error
+	ver      Version
+	txIn     TxnInput
+	txOut    TxnOutput
+	txAllOut TxnOutput
+}
+
+type ExecTask interface {
+	Execute(mvh *MVHashMap, incarnation int) error
+	MVReadList() []ReadDescriptor
+	MVWriteList() []WriteDescriptor
+	MVFullWriteList() []WriteDescriptor
```

Comment:
Added a new write set that represent the full write list. `MVFullWriteList` will be a parent set of `MVWriteList`. This is needed because we had to store state objects in MVHashmap (the records whose key is only an address without any subpath), but we don't want to detect it as a write conflict during validation. Therefore, in validation, only `MVWriteList` will be checked, while when recording writes to MVHashmap upon the completion of a task, we will write MVFullWriteList to MVHashmap.